backend/src/apiserver/server/util.go

```diff
@@ -289,110 +290,55 @@ func getUserIdentityFromHeader(userIdentityHeader, prefix string) (string, error
-		// Skip authz if not multi-user mode.
+	if common.IsMultiUserSharedReadMode() &&
+		(resourceAttributes.Verb == common.RbacResourceVerbGet ||
+			resourceAttributes.Verb == common.RbacResourceVerbList) {
+		glog.Infof("Multi-user shared read mode is enabled. Request allowed: %+v", resourceAttributes)
 		return nil
 	}
 
-	namespace := common.GetNamespaceFromAPIResourceReferences(resourceRefs)
-	if len(namespace) == 0 {
-		return util.NewBadRequestError(errors.New("Namespace required in Kubeflow deployment for authorization."), "Namespace required in Kubeflow deployment for authorization.")
-	}
-	err := isAuthorized(resourceManager, ctx, namespace)
-	if err != nil {
-		return util.Wrap(err, "Failed to authorize with API resource references")
-	}
-	return nil
-}
-
-func CanAccessNamespace(resourceManager *resource.ResourceManager, ctx context.Context, namespace string) error {
-	if common.IsMultiUserMode() == false {
-		// Skip authz if not multi-user mode.
-		return nil
-	}
-
-	if len(namespace) == 0 {
-		return util.NewBadRequestError(errors.New("Namespace required for authorization."), "Namespace required for authorization.")
-	}
-	err := isAuthorized(resourceManager, ctx, namespace)
-	if err != nil {
-		return util.Wrap(err, "Failed to authorize namespace")
-	}
-	return nil
-}
-
-// isAuthorized verified whether the user identity, which is contains in the context object,
-// can access the target namespace. If the returned error is nil, the authorization passes.
-// Otherwise, Authorization fails with a non-nil error.
-func isAuthorized(resourceManager *resource.ResourceManager, ctx context.Context, namespace string) error {
+	glog.Info("Getting user identity...")
 	userIdentity, err := getUserIdentity(ctx)
 	if err != nil {
 		return util.Wrap(err, "Bad request.")
 	}
 
 	if len(userIdentity) == 0 {
-		return util.NewBadRequestError(errors.New("Request header error: user identity is empty."), "Request header error: user identity is empty.")
+		return util.NewUnauthenticatedError(errors.New("Request header error: user identity is empty."), "Request header error: user identity is empty.")
 	}
 
-	isAuthorized, err := resourceManager.IsRequestAuthorized(userIdentity, namespace)
+	glog.Infof("User: %s, ResourceAttributes: %+v", userIdentity, resourceAttributes)
+	glog.Info("Authorizing request...")
+	err = resourceManager.IsRequestAuthorized(userIdentity, resourceAttributes)
 	if err != nil {
-		return util.Wrap(err, "Authorization failure.")
-	}
-
-	if isAuthorized == false {
-		glog.Infof("Unauthorized access for %s to namespace %s", userIdentity, namespace)
-		return util.NewBadRequestError(errors.New("Unauthorized access for "+userIdentity+" to namespace "+namespace), "Unauthorized access for "+userIdentity+" to namespace "+namespace)
+		glog.Info(err.Error())
+		return err
 	}
 
-	glog.Infof("Authorized user %s in namespace %s", userIdentity, namespace)
+	glog.Infof("Authorized user '%s': %+v", userIdentity, resourceAttributes)
```

Comment:
These authorization logs will be very common, every request comes with it.
Shall we change log level to `trace` or `debug`?

For most of the cases, users should depend on error message for debugging, rather than api server logs.

Reply:
I can only see `Info`, `Warning`, `Error`, and `Fatal` levels. Can you help me how can I do `debug` or sth different?
Here's what I found in the repo, not sure if it helps: https://github.com/golang/glog/blob/23def4e6c14b4da8ac2ed8007337bc5eb5007998/glog.go#L91-L115

Reply:
OK, sorry I thought there were more. Then let's go with info for now.